subsys/bluetooth/mesh/prov_device.c

```diff
@@ -42,11 +42,22 @@
 #include "prov.h"
 #include "settings.h"
 
+static struct {
+	uint8_t net_key[16];
+	uint16_t net_idx;
+	uint8_t flags;
+	uint32_t iv_index;
+	uint16_t addr;
+	uint8_t dev_key[16];
+} __packed * prov_data_ptr;
```

Comment:
There is no reason to have this structure as a packed one. The compiler will generate safer code to get access to all members. That means all data that can be accessible via halfword\word instructions will get access via byte instructions. That causes unnecessary havier final code. Packed data is better to use for serialization\desearilization purposes.

Reply:
Did I think there might be a potential alignment problem?

Reply:
Do we force this structure to point to a byte aligned space? Should we consider alignment?